File: tests/cases/general/__snapshots__/parity-array.lua
local ____exports = {}
function ____exports.__main()
    local arr = {10, 20, 30, 40, 50}
    local sum = 0
    local i = 0
    while true do
        do
            if not (i < #arr) then
                break
            end
            sum = sum + arr[i + 1]
            i = i + 1
        end
        ::____continue_0::
    end
    return {sum = sum, len = #arr, first = arr[1], last = arr[#arr - 1 + 1]}
end
return ____exports

File: tests/cases/general/parity-array.ts
// Parity demo: array indexing + length. Exercises the EsArrayLength /
// EsIndex divergences (DIV-ARR-INDEX-001) end-to-end, and tests that
// dunder's 0-to-1 index translation produces the same observable
// results as JS.

export function __main() {
  const arr = [10, 20, 30, 40, 50];
  let sum = 0;
  let i = 0;
  while (i < arr.length) {
    sum = sum + arr[i];
    i = i + 1;
  }
  return { sum, len: arr.length, first: arr[0], last: arr[arr.length - 1] };
}
